src/bsp_types/mappings/to_publish_diagnostics.rs

```diff
@@ -47,14 +36,21 @@ enum MappedRustChildDiagnostic {
 ///    `relatedInformation` or additional message lines.
 ///
 /// If the diagnostic has no primary span this will return `None`
-fn map_cargo_diagnostic_to_bsp(
+pub fn map_cargo_diagnostic_to_bsp(
```

Comment:
do zrobienia kiedyś - refactor tej funkcji żeby to nie było takie spaghetti code